spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -82,16 +78,16 @@ private void receiveMessage(PubsubMessage pubsubMessage, AckReplyConsumer consum
 		}
 
 		try {
-			Message<?> internalMessage =
-					this.messageConverter == null
-							? MessageBuilder.withPayload(pubsubMessage.getData().toByteArray())
-							.copyHeaders(messageHeaders)
-							.build()
-							: this.messageConverter.toMessage(
-									pubsubMessage.getData().toStringUtf8(),
-							new MessageHeaders(messageHeaders));
-
-			sendMessage(internalMessage);
+			Message message;
+			if (this.messageConverter != null) {
+				message = this.messageConverter.toMessage(pubsubMessage.getData().toByteArray(),
```

Comment:
@artembilan Maybe we should pass in the whole `PubsubMessage` to the converter instead of just its payload?

Reply:
No, that's not, because that way the whole `PubsubMessage` is going to be as a payload of the Spring message to send. Just take a look into the `MappingJackson2MessageConverter.convertToInternal()`. We do there `this.objectMapper.writeValue(generator, payload);`, I'm not sure that everybody is going to be happy that you try to convert the whole `PubsubMessage` there...

Reply:
OK, thanks! Any remaining concerns about the PR?